reference/supervisor.md

```diff
@@ -122,28 +163,335 @@ If no node is currently selected, the function returns NULL.
 #include <webots/supervisor.h>
 
+#define WB_NODE_NO_NODE
+/* VRML97 nodes */
+#define WB_NODE_APPEARANCE
+#define WB_NODE_BACKGROUND
+#define WB_NODE_BOX
+#define WB_NODE_COLOR
+#define WB_NODE_COMPOSED_CUBE_MAP_TEXTURE
+#define WB_NODE_COMPOSED_SHADER
+#define WB_NODE_CONE
+#define WB_NODE_COORDINATE
+#define WB_NODE_CYLINDER
+#define WB_NODE_DIRECTIONAL_LIGHT
+#define WB_NODE_ELEVATION_GRID
+#define WB_NODE_EXTRUSION
+#define WB_NODE_FOG
+#define WB_NODE_GROUP
+#define WB_NODE_IMAGE_TEXTURE
+#define WB_NODE_INDEXED_FACE_SET
+#define WB_NODE_INDEXED_LINE_SET
+#define WB_NODE_LIDAR
+#define WB_NODE_MATERIAL
+#define WB_NODE_MULTI_TEXTURE
+#define WB_NODE_MUSCLE
+#define WB_NODE_POINT_LIGHT
+#define WB_NODE_SHADER_PART
+#define WB_NODE_SHAPE
+#define WB_NODE_SPHERE
+#define WB_NODE_SPOT_LIGHT
+#define WB_NODE_SWITCH
+#define WB_NODE_TEXTURE_COORDINATE
+#define WB_NODE_TEXTURE_TRANSFORM
+#define WB_NODE_TRACK
+#define WB_NODE_TRACK_WHEEL
+#define WB_NODE_TRANSFORM
+#define WB_NODE_UNIFORM
+#define WB_NODE_VIEWPOINT
+#define WB_NODE_WORLD_INFO
+/* non-vrml geometry */
+#define WB_NODE_CAPSULE
+#define WB_NODE_PLANE
+/* robots */
+#define WB_NODE_ROBOT
+#define WB_NODE_SUPERVISOR
+#define WB_NODE_DIFFERENTIAL_WHEELS
+/* misc */
+#define WB_NODE_BALL_JOINT
+#define WB_NODE_BALL_JOINT_PARAMETERS
+#define WB_NODE_CHARGER
+#define WB_NODE_CONTACT_PROPERTIES
+#define WB_NODE_DAMPING
+#define WB_NODE_FLUID
+#define WB_NODE_FOCUS
+#define WB_NODE_HINGE_JOINT
+#define WB_NODE_HINGE_JOINT_PARAMETERS
+#define WB_NODE_HINGE_2_JOINT
+#define WB_NODE_HINGE_2_JOINT_PARAMETERS
+#define WB_NODE_IMMERSION_PROPERTIES
+#define WB_NODE_JOINT_PARAMETERS
+#define WB_NODE_LENS
+#define WB_NODE_LENS_FLARE
+#define WB_NODE_PHYSICS
+#define WB_NODE_RECOGNITION
+#define WB_NODE_SLIDER_JOINT
+#define WB_NODE_SOLID
+#define WB_NODE_SOLID_REFERENCE
+#define WB_NODE_SLOT
+#define WB_NODE_ZOOM
+/* devices */
+#define WB_NODE_ACCELEROMETER
+#define WB_NODE_BRAKE
+#define WB_NODE_CAMERA
+#define WB_NODE_COMPASS
+#define WB_NODE_CONNECTOR
+#define WB_NODE_DISPLAY
+#define WB_NODE_DISTANCE_SENSOR
+#define WB_NODE_EMITTER
+#define WB_NODE_GPS
+#define WB_NODE_GYRO
+#define WB_NODE_INERTIAL_UNIT
+#define WB_NODE_LED
+#define WB_NODE_LIGHT_SENSOR
+#define WB_NODE_LINEAR_MOTOR
+#define WB_NODE_MICROPHONE
+#define WB_NODE_PEN
+#define WB_NODE_POSITION_SENSOR
+#define WB_NODE_PROPELLER
+#define WB_NODE_RADAR
+#define WB_NODE_RADIO
+#define WB_NODE_RANGE_FINDER
+#define WB_NODE_ROTATIONAL_MOTOR
+#define WB_NODE_RECEIVER
+#define WB_NODE_SKIN
+#define WB_NODE_SPEAKER
+#define WB_NODE_TOUCH_SENSOR
+
 WbNodeType wb_supervisor_node_get_type(WbNodeRef node);
 const char *wb_supervisor_node_get_type_name(WbNodeRef node);
 const char *wb_supervisor_node_get_base_type_name(WbNodeRef node);
 ```
 
+%tab-end
+
+%tab "C++"
+
+```cpp
+#include <webots/Node.hpp>
+
+namespace webots {
+  class Node {
+    enum {
+      NO_NODE, ACCELEROMETER, APPEARANCE, BACKGROUND, BALL_JOINT, BALL_JOINT_PARAMETERS,
```

Comment:
The list of nodes is not up-to-date (same for Java and Python) but we can update it on a different PR.

missing:
* WB_NODE_LENS
* WB_NODE_LENS_FLARE
* WB_NODE_MUSCLE
* WB_NODE_MULTI_TEXTURE
* WB_NODE_COMPOSED_CUBE_MAP_TEXTURE
* WB_NODE_COMPOSED_SHADER
* WB_NODE_SKIN
* WB_NODE_UNIFORM

to be removed:
* WB_LENS_DISTORTION

Reply:
I would prefer in another PR, it would be time to recheck all this topic, and possibly add tests for this.